### DISCUSSION TITLE
what is first neighbour and second neighbour as implemented in Mobi?

### DISCUSSION THREAD

**Discussion by hainiwen** [Open-Systems-Pharmacology/Forum#1516] [Questions & Problems] [answered]
Hi, 
So sometimes i meet with 'first neighbour' or 'second neighbour' in the path of parameters. Is it kind of default setting in Mobi? What are the general rules like? I am a bit lost when the model is big and I try to understand which parameter is now referring too...

Haini

**Comment by moPellowe** [comment, marked as answer]
FIRST_NEIGHBOR and SECOND_NEIGHBOR refer to the "beginning" and "end" of a neighborhood. When you create a neighborhood in a spatial structure, you link from one compartment to another, i.e. from FIRST_NEIGHBOR to SECOND_NEIGHBOR.

**Reply by msevestre** [reply]
To add on top of @moPellowe's response, when you have a parameter in a neighborhood defined by a formula, you can access property of connected container by using the keyword FIRST_NEiGHBOR. The actual neighbor will be resolved in the simulation creation process.
For example, FIRST_NEiGHBOR in a neighborhood between kidney/pls and venous blood /pls will be kidney/pls. From there, you can construct path relative to this resolved neighbor. 
FIRST_NEIGHBOR/.. would be kidney 
FIRST_NEIGHBOR/../Volume would be volume of kidney etc..

**Reply by hainiwen** [reply]
thanks!!!! now it makes more sense to me!!
